app/kubernetes-ingress-controller/2.1.x/concepts/ha-and-scaling.md

```diff
@@ -0,0 +1,69 @@
+---
+title: High-availability and Scaling
+---
+
+## High availability
+
+The {{site.kic_product_name}} is designed to be reasonably easy to operate and
+be highly available, meaning, when some expected failures do occur, the
+Controller should be able to continue to function with minimum possible
+service disruption.
+
+The {{site.kic_product_name}} is composed of two parts: 1. Kong, which handles
+the requests, 2. Controller, which configures Kong dynamically.
+
+Kong itself can be deployed in a Highly available manner by deploying
+multiple instances (or pods). Kong nodes are state-less, meaning a Kong pod
+can be terminated and restarted at any point of time.
+
+The controller itself can be stateful or stateless, depending on if a database
+is being used or not.
+
+If a database is not used, then the Controller and Kong are deployed as
+colocated containers in the same pod and each controller configures the Kong
+container that it is running with.
+
+For cases when a database is necessary, the Controllers can be deployed
+on multiple zones to provide redundancy. In such a case, a leader election
+process will elect one instance as a leader, which will manipulate Kong's
+configuration.
+
+### Leader election
+
+The {{site.kic_product_name}} performs a leader-election when multiple
+instances of the controller are running to ensure that only a single Controller
+is actively pushing changes to Kong's database (when running in DB-mode).
+If multiple controllers are making changes to the database, it is possible that
+the controllers step over each other.
+If an instance of the controller fails, any other container which is a follower,
+takes up the leadership and then continues syncing Kong's configuration from
+Kubernetes.
+
```

Comment:
```suggestion
Kubernetes.

When not running in DBLESS mode leader election is automatically disabled as each
instance of the controller will correspond with a distinct instance of the gateway.

```

Reply:
This is covered in already in the first paragraph of the section:

> The {{site.kic_product_name}} performs a leader-election when multiple    
> instances of the controller are running to ensure that only a single Controller    
> is actively pushing changes to Kong's database (when running in DB-mode).

Reply:
I gotcha, to me I suppose I had some difficultly parsing this language and got tripped up. I would be in favor of some rewording here.

Reply:
updated header and rewritten first paragraph in 26202c9666 clearer? Not exactly sure what was unclear originally.